simple-rest-server/checkout.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+)
+
+func checkTaskAsComplete(taskID string) (err error) {
+	// Get the index.
+	index, err = getIndexByTaskID(taskID)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	// Lock the tasks.
+	accessTasks.Lock()
+	defer accessTasks.Unlock()
+
+	// Set the task to checked.
+	allTasks[index].checked = true
```

Comment:
You shouldn't defer the `accessTasks.Unlock()` and move it after this line for readability and preventing bad usage.